pkg/controller/dataimportcron-controller.go

```diff
@@ -733,7 +727,14 @@ func (r *DataImportCronReconciler) handleSnapshot(ctx context.Context, dataImpor
 			return err
 		}
 	} else {
-		if cc.IsSnapshotReady(currentSnapshot) {
+		updatedSnapshot := currentSnapshot.DeepCopy()
+		cc.CopyAllowedLabels(pvc.GetLabels(), updatedSnapshot, true)
+		if !reflect.DeepEqual(currentSnapshot, updatedSnapshot) {
+			if err := r.client.Update(ctx, updatedSnapshot); err != nil {
+				return err
+			}
+		}
```

Comment:
I don't think this would solve the case for an existing snapshots with no labels, maybe we could just do something similar to https://github.com/kubevirt/containerized-data-importer/pull/3155/commits/755fe0bc6a612b55c426b3186a51ea12e5274ce2 by copying the labels from the data source to the snapshot. This would only be needed for a brief time window before the cron gets a sha update.

Reply:
Updated the approach to match 755fe0b, PTAL!